app/src/main/java/com/example/myapplication/ThirdFragment.java

```diff
@@ -87,52 +78,56 @@ public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
             UIHelpers.lightDark(v, UIHelpers.darkMode);
 
         });
-        binding.switch1.setOnCheckedChangeListener((buttonView, isChecked) -> setSwitchColor(binding.switch1));
-        binding.switch2.setOnCheckedChangeListener((buttonView, isChecked) -> setSwitchColor(binding.switch2));
+        binding.switch1.setOnCheckedChangeListener((buttonView, isChecked) -> {
+            setSwitchColor(binding.switch1);
+            UserModel.getMatchData().setWorkingAuto(isChecked);
+        });
+        binding.switch2.setOnCheckedChangeListener((buttonView, isChecked) -> {
+            setSwitchColor(binding.switch2);
+            UserModel.getMatchData().setMoveOutOfZone(isChecked);
+        });
         binding.switch3.setOnCheckedChangeListener((buttonView, isChecked) -> setSwitchColor(binding.switch3));
         binding.toSecondFragment.setOnClickListener(view12 -> {
             NavHostFragment.findNavController(ThirdFragment.this)
                     .navigate(R.id.action_ThirdFragment_to_SecondFragment);
-            //getting the value of variables when switching to teleop from auto
-            /*binding.switch1.isChecked();
-            binding.switch2.isChecked();
-            binding.switch3.isChecked();
-            numNotes.getText();
-            numNotesInAmp.getText();
-            Contact.getSelectedItem();*/
         });
-        String[] typeContact = {"No contact", "Teammate to Teammate", "Teammate to opponent"};
-        ArrayAdapter<String> contact = new ArrayAdapter<>(view.getContext(), android.R.layout.simple_spinner_dropdown_item, typeContact);
-        binding.spinner.setAdapter(contact);
-        binding.spinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+        UIHelpers.lightDark(v, UIHelpers.darkMode);
+
+        binding.numNotes.getBackground().setColorFilter(Color.parseColor("#73C2F0"), PorterDuff.Mode.SRC_ATOP);
```

Comment:
The call `.getBackground().setColorFilter(Color.parseColor("#73C2F0"), PorterDuff.Mode.SRC_ATOP)` is repeated here; consider making a helper method (e.g., `updateNumberBackgroundControl`)